src/test/java/com/transferwise/openbanking/client/oauth/PrivateKeyJwtAuthenticationTest.java

```diff
@@ -71,4 +79,69 @@ private AspspDetails aAspspDefinition() {
             .signingAlgorithm(AlgorithmIdentifiers.RSA_PSS_USING_SHA256)
             .build();
     }
+
+    @Test
+    void addClientAuthentication_defaultAspsDetails() throws Exception {
+        // Given
+        GetAccessTokenRequest getAccessTokenRequest = new GetAccessTokenRequest();
+        AspspDetails aspspDetails = TestAspspDetails.builder()
+            .clientId("aClientId")
+            .privateKeyJwtAuthenticationAudience("/autenticationAudienceUrl")
+            .signingAlgorithm(AlgorithmIdentifiers.RSA_PSS_USING_SHA256)
+            .build();
+
+        when(jwtClaimsSigner.createSignature(any(JwtClaims.class), any(AspspDetails.class))).thenReturn("aJwtSignature");
+
+        // When
+        privateKeyJwtAuthentication.addClientAuthentication(getAccessTokenRequest, aspspDetails);
+
+        // Then
+        assertEquals("urn:ietf:params:oauth:client-assertion-type:jwt-bearer", getAccessTokenRequest.getRequestBody().get("client_assertion_type"));
+        assertEquals("aJwtSignature", getAccessTokenRequest.getRequestBody().get("client_assertion"));
+
+        ArgumentCaptor<JwtClaims> jwtClaimsCaptor = ArgumentCaptor.forClass(JwtClaims.class);
+        verify(jwtClaimsSigner).createSignature(jwtClaimsCaptor.capture(), eq(aspspDetails));
+
+        JwtClaims jwtClaims = jwtClaimsCaptor.getValue();
+        assertEquals("aClientId", jwtClaims.getIssuer());
+        assertEquals("aClientId", jwtClaims.getSubject());
+        assertEquals(1, jwtClaims.getAudience().size());
+        assertEquals("/autenticationAudienceUrl", jwtClaims.getAudience().get(0));
+        assertNotNull(jwtClaims.getIssuedAt());
+        assertNotNull(jwtClaims.getExpirationTime());
+        assertNotNull(jwtClaims.getJwtId());
```

Comment:
there is some duplicated code here, I could eventually move it to a specific method...